mlir/lib/Dialect/GPU/Transforms/EliminateBarriers.cpp

```diff
@@ -424,27 +422,18 @@ static bool maybeCaptured(Value v) {
 /// everything. This seems sufficient to achieve barrier removal in structured
 /// control flow, more complex cases would require a proper dataflow analysis.
 static bool mayAlias(Value first, Value second) {
-  DEBUG_WITH_TYPE(DEBUG_TYPE_ALIAS, {
-    DBGS_ALIAS() << "checking aliasing between ";
-    DBGS_ALIAS() << first << "\n";
-    DBGS_ALIAS() << "                      and ";
-    DBGS_ALIAS() << second << "\n";
-  });
+  LDBG(1, DEBUG_TYPE_ALIAS)
```

Comment:
I know this is following default arguments are last kind of C++ style, but I think the opposite reads nicer here:

```
LDBG(DEBUG_TYPE, 1)
```

As the former is the higher-level filter and the latter the level within that. Thinking out loud, I wonder: could we even overload on type here? (so `LDBG(DEBUG_TYPE)` could work too).

Reply:
Let me look at the overload idea, I like it :)

I wanted to preserve the more common case where you have a single DEBUG_TYPE in a file and you just want to adjust for the log level `LDBG(2)`  instead of having to be explicit about the DEBUG_TYPE just to change the log level.